src/bridge/SequencerInbox.sol

```diff
@@ -245,48 +289,19 @@ contract SequencerInbox is GasRefundEnabled, ISequencerInbox {
         if (!isBatchPoster[msg.sender] && msg.sender != address(rollup)) revert NotBatchPoster();
         (bytes32 dataHash, IBridge.TimeBounds memory timeBounds) = formDataHash(
             data,
-            afterDelayedMessagesRead
+            afterDelayedMessagesRead,
+            IBridge.BatchDataLocation.SeparateBatchEvent
         );
-        uint256 seqMessageIndex;
-        {
-            // Reformat the stack to prevent "Stack too deep"
-            uint256 sequenceNumber_ = sequenceNumber;
-            IBridge.TimeBounds memory timeBounds_ = timeBounds;
-            bytes32 dataHash_ = dataHash;
-            uint256 afterDelayedMessagesRead_ = afterDelayedMessagesRead;
-            uint256 prevMessageCount_ = prevMessageCount;
-            uint256 newMessageCount_ = newMessageCount;
-            // we set the calldata length posted to 0 here since the caller isn't the origin
-            // of the tx, so they might have not paid tx input cost for the calldata
-            seqMessageIndex = addSequencerL2BatchImpl(
-                dataHash_,
-                afterDelayedMessagesRead_,
-                0,
-                prevMessageCount_,
-                newMessageCount_,
-                timeBounds_,
-                IBridge.BatchDataLocation.SeparateBatchEvent
-            );
-            if (seqMessageIndex != sequenceNumber_ && sequenceNumber_ != ~uint256(0))
-                revert BadSequencerNumber(seqMessageIndex, sequenceNumber_);
-        }
-        emit SequencerBatchData(seqMessageIndex, data);
-    }
-
-    modifier validateBatchData(bytes calldata data) {
-        uint256 fullDataLen = HEADER_LENGTH + data.length;
-        if (fullDataLen > maxDataSize) revert DataTooLarge(fullDataLen, maxDataSize);
-        if (data.length > 0 && (data[0] & DATA_AUTHENTICATED_FLAG) == DATA_AUTHENTICATED_FLAG) {
-            revert DataNotAuthenticated();
-        }
-        // the first byte is used to identify the type of batch data
-        // das batches expect to have the type byte set, followed by the keyset (so they should have at least 33 bytes)
-        if (data.length >= 33 && data[0] & 0x80 != 0) {
-            // we skip the first byte, then read the next 32 bytes for the keyset
-            bytes32 dasKeysetHash = bytes32(data[1:33]);
-            if (!dasKeySetInfo[dasKeysetHash].isValidKeyset) revert NoSuchKeyset(dasKeysetHash);
-        }
-        _;
+        addSequencerL2BatchImpl(
+            sequenceNumber,
+            dataHash,
+            timeBounds,
+            afterDelayedMessagesRead,
+            prevMessageCount,
+            newMessageCount,
+            IBridge.BatchDataLocation.SeparateBatchEvent
+        );
+        emit SequencerBatchData(sequenceNumber, data);
```

Comment:
Did you look at how the first 4844 PR worked? It emitted the full data that L2 would read, including the data hashes and such, which simplifies the node's inbox reader a bit. It also means you don't need the new batch data location.

Reply:
Yeh, I did see the PR. I need to take a look in the inbox reader I guess, but dont you have access to the full transaction data there, including the blob versioned hashes?

Reply:
I assuming you mean this for the original PR: https://github.com/OffchainLabs/nitro-contracts/compare/develop...eip-4844-contracts

Reply:
Yeah I think so, cc @Tristan-Wilson would know better than me